src/main/java/org/springframework/data/elasticsearch/support/ReactiveSupport.java

```diff
@@ -15,25 +15,16 @@
  */
 package org.springframework.data.elasticsearch.support;
 
-import java.util.concurrent.atomic.AtomicBoolean;
-
-import org.springframework.data.repository.util.ClassUtils;
+import org.springframework.util.ClassUtils;
 
 /**
  * @author Peter-Josef Meisch
  * @since 4.0
  */
 public final class ReactiveSupport {
-	private ReactiveSupport() {}
 
-	/**
-	 * @return true if project reactor is on the classpath
-	 */
-	public static boolean isReactorAvailable() {
-		AtomicBoolean available = new AtomicBoolean(false);
-		ClassUtils.ifPresent("reactor.core.publisher.Flux", null, aClass -> {
-			available.set(true);
-		});
-		return available.get();
-	}
+	public static boolean PROJECT_REACTOR_AVAILABLE = ClassUtils.isPresent("reactor.core.publisher.Mono",
+			ReactiveSupport.class.getClassLoader());
+
+	private ReactiveSupport() {}
```

Comment:
This class could be entirely removed in favor of `org.springframework.data.repository.util.ReactiveWrappers` (`ReactiveWrappers.isAvailable(ReactiveLibrary.PROJECT_REACTOR)`).

Reply:
I would have used that, but the `ReactiveLibrary` enum is not public accessible.

Reply:
Thanks for making us aware. I filed https://jira.spring.io/browse/DATACMNS-1679 to fix that issue.